src/InsertionsClient/Api/Props/PropsVariableDeducer.cs

```diff
@@ -0,0 +1,175 @@
+
+            nugetDownloadBlock.LinkTo(filenameMatchBlock, new DataflowLinkOptions() { PropagateCompletion = true });
+
+            foreach (PackageUpdateResult package in updatedPackages)
+            {
+                nugetDownloadBlock.Post(package);
+            }
+
+            nugetDownloadBlock.Complete();
+            filenameMatchBlock.Completion.Wait();
+
+            return deducedVariables.ToList();
+        }
+
+        private async Task<IEnumerable<string>> GetPackageFileList(DefaultConfigUpdater defaultConfigUpdater, PackageUpdateResult packageUpdate)
+        {
+            string? link = defaultConfigUpdater.GetPackageLink(packageUpdate.PackageId);
+
+            if (link == null)
+            {
+                // We don't know where nuget package should be extracted to.
+                return Enumerable.Empty<string>();
+            }
+
+            SourceCacheContext cache = new SourceCacheContext();
+            CancellationTokenSource cancellationToken = new CancellationTokenSource();
+            NuGetVersion packageVersion = new NuGetVersion(packageUpdate.NewVersion);
+
+            // Stream that will receive the package bytes
+            using MemoryStream packageStream = new MemoryStream();
+
+            try
+            {
+                SourceRepository? rep = Repository.Factory.GetCoreV3(_feed, FeedType.HttpV3);
+                if (_credentials != null)
+                {
+                    rep.PackageSource.Credentials = _credentials;
+                }
+
+                FindPackageByIdResource? resource = await rep.GetResourceAsync<FindPackageByIdResource>();
+
+                Trace.WriteLine($"Downloading package {packageUpdate.PackageId}-{packageUpdate.NewVersion} into memory.");
+
+                bool downloadResult = await resource.CopyNupkgToStreamAsync(
+                    packageUpdate.PackageId,
+                    packageVersion,
+                    packageStream,
+                    cache,
+                    NullLogger.Instance,
+                    cancellationToken.Token);
+
+                if (!downloadResult)
+                {
+                    Trace.WriteLine($"There is an issue downloading nuget package: {packageUpdate.PackageId}");
+                    return Enumerable.Empty<string>();
+                }
+
+                Trace.WriteLine($"Downloading complete for package {packageUpdate.PackageId}.");
+
+                using PackageArchiveReader packageReader = new PackageArchiveReader(packageStream);
+
+                return packageReader.GetFiles().Select(file =>
+                {
+                    // Construct path to the file
+                    // Link always has backslash. Make sure it is consistent.
+                    return link + "\\" + file.Replace('/', '\\');
+                }).ToList();
+            }
+            catch (Exception e)
+            {
+                Trace.WriteLine("There is an issue downloading nuget package: " + e.ToString());
```

Comment:
consider a homogeneous way of reporting e to the Trace.  I've seen `Trace.WriteLine(e);` earlier in the review

Reply:
Changed to ensure homogeneity.